ksql-engine/src/test/resources/query-validation-tests/group-by.json

```diff
@@ -46,10 +46,54 @@
       ]
     },
     {
-      "name": "group by field with re-key",
+      "name": "with groupings (stream->table)",
+      "statements": [
+        "CREATE STREAM TEST (f1 INT, f2 VARCHAR, f3 INT) WITH (kafka_topic='test_topic', KEY='f1', value_format='DELIMITED');",
+        "CREATE TABLE OUTPUT AS SELECT f1, f2, COUNT(*) FROM TEST GROUP BY f3, (f2, f1);"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": 1, "value": "1,a,-1"},
+        {"topic": "test_topic", "key": 2, "value": "2,b,-2"},
+        {"topic": "test_topic", "key": 1, "value": "1,a,-1"},
+        {"topic": "test_topic", "key": 2, "value": "2,b,-2"},
+        {"topic": "test_topic", "key": 3, "value": "3,a,-3"}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": "-1|+|a|+|1", "value": "1,a,1"},
+        {"topic": "OUTPUT", "key": "-2|+|b|+|2", "value": "2,b,1"},
+        {"topic": "OUTPUT", "key": "-1|+|a|+|1", "value": "1,a,2"},
+        {"topic": "OUTPUT", "key": "-2|+|b|+|2", "value": "2,b,2"},
+        {"topic": "OUTPUT", "key": "-3|+|a|+|3", "value": "3,a,1"}
+      ]
+    },
+    {
+      "name": "fields (table->table)",
+      "statements": [
+        "CREATE TABLE TEST (f1 INT, f2 VARCHAR) WITH (kafka_topic='test_topic', KEY='f1', value_format='DELIMITED');",
+        "CREATE TABLE OUTPUT AS SELECT f1, f2, COUNT(*) FROM TEST GROUP BY f2, f1;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": 1, "value": "1,a"},
+        {"topic": "test_topic", "key": 2, "value": "2,b"},
+        {"topic": "test_topic", "key": 1, "value": "1,b"},
+        {"topic": "test_topic", "key": 2, "value": null},
+        {"topic": "test_topic", "key": 1, "value": "1,a"}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": "a|+|1", "value": "1,a,1"},
+        {"topic": "OUTPUT", "key": "b|+|2", "value": "2,b,1"},
+        {"topic": "OUTPUT", "key": "a|+|1", "value": "1,a,0"},
+        {"topic": "OUTPUT", "key": "b|+|1", "value": "1,b,1"},
+        {"topic": "OUTPUT", "key": "b|+|2", "value": "2,b,0"},
+        {"topic": "OUTPUT", "key": "b|+|1", "value": "1,b,0"},
```

Comment:
To make sure that I wrap my head around this - the reason we reduce the count to for group (b,1) to 0 is because a new _table_ update for key 1 comes in that updates the value of f2 from "b" to "a"?

Reply:
Correct.